std/experimental/ndslice/algorithm.d

```diff
@@ -0,0 +1,1045 @@
+/**
+$(SCRIPT inhibitQuickIndex = 1;)
+
+This is a submodule of $(MREF std, experimental, ndslice).
+It contains generic multidimensional iteration algorithms.
+
+$(BOOKTABLE Cheat Sheet,
+$(TR $(TH Operator Name) $(TH Type) $(TH Functions / Seeds #) $(TH Tensors #) $(TH Returns) $(TH First Argument))
+$(T6 ndMap, Lazy, `>=1`, `1`, Tensor, Tensor)
+$(T6 ndFold, Eagerly, `>=1`, `1`, Scalar, Tensor)
+$(T6 ndReduce, Eagerly, `1`, `>=1`, Scalar, Seed)
+$(T6 ndEach, Eagerly, `0`, `>=1`, `void`, Tensor)
+)
+
+All operators are suitable to change tensors using `ref` argument qualification in a function declaration.
+
+Note:
+    $(SUBREF iteration, transposed) and
+    $(SUBREF selection, pack) can be used to specify dimensions.
+
+License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors:   Ilya Yaroshenko
+
+Source:    $(PHOBOSSRC std/_experimental/_ndslice/_algorithm.d)
+
+Macros:
+SUBREF = $(REF_ALTTEXT $(TT $2), $2, std,experimental, ndslice, $1)$(NBSP)
+T2=$(TR $(TDNW $(LREF $1)) $(TD $+))
+T6=$(TR $(TDNW $(LREF $1)) $(TD $2) $(TD $3) $(TD $4) $(TD $5) $(TD $6))
+*/
+module std.experimental.ndslice.algorithm;
+
+import std.traits;
+import std.meta;
+
+import std.experimental.ndslice.internal;
+import std.experimental.ndslice.slice;
+
+private template TensorFronts(size_t length)
```

Comment:
Hmm, it requires to import `std.format` :-(

The number of dimensions is not very large. Common numbers are 2 or 3.